ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/drivers/modifiers/ModifierFactory.java

```diff
@@ -85,15 +86,19 @@ public static synchronized ConnectionModifier getModifier (String dbName, int ma
 	    if (s.equalsIgnoreCase(dbName + "_-1_-1"))
 		driverMatch = s;
 	}
+	ConnectionModifier modifier = defaultIsSameRMOverride ? isSameRMModifier : null;
 
-	if (exactMatch != null)
-		return _modifiers.get(exactMatch);
-	if (majorMatch != null)
-		return _modifiers.get(majorMatch);
 	if (driverMatch != null)
-		return _modifiers.get(driverMatch);
+		modifier = _modifiers.get(driverMatch);
+	if (majorMatch != null)
+		modifier = _modifiers.get(majorMatch);
+	if (exactMatch != null)
+		modifier = _modifiers.get(exactMatch);
 
-	return defaultIsSameRMOverride ? isSameRMModifier : null;
+	if (jdbcLogger.logger.isTraceEnabled()) {
+		jdbcLogger.logger.trace("ConnectionModifier for: " + modifier.getClass().getName() + " for " + dbName + " " + major + "/" + minor + " (defaultIsSameRMOverride was " + defaultIsSameRMOverride + ")");
```

Comment:
I think there is still possibility that `modifier` will be null. I mean when either `if` case match then it could happen and `NullPointerException` could be thrown from the `logger.trace` call.

Am I wrong?

Reply:
You are not wrong - in fact you are right :) I will adjust